src/com/lilithsthrone/game/character/GameCharacter.java

```diff
@@ -2887,8 +3007,421 @@ public boolean isSlave() {
+			}
+			
+			Main.game.getPlayer().incrementObedience(obedienceChangeTotal, false);
+			this.timeOfNextRuleCheck += 60;
+			
+			// Checking to see if the player deserves a punishment for being a bad [player.gender] ;3
+			if(this.obedience < GameCharacter.FORCED_PUNISHMENT_THRESHOLD || obedienceChangeTotal < GameCharacter.FORCED_PUNISHMENT_DROP_TRIGGER)
+			{
+				forcePunishment = true;
+			}
+		}
+		
+		// Checking for event spawning here
+		if(this.timeOfNextEventCheck <= minutesPassed && Main.game.getPlayer().isWithinOwnersPropery() && Main.game.getActiveNPC() == null)
+		{
+			
+			// At 0 weight, an event won't be generated. If all events return 0 weight, then no event happens.
+			Main.game.setActiveNPC((NPC)this);
+			int currentCandidateWeight = 0;
+			PlayerSlaveryEvent currentCandidate = null;
+			
+			if(this.getSlaveryEvents() != null)
+			{
+				for(PlayerSlaveryEvent slaveryEvent : this.getSlaveryEvents()) {
+					if(!forcePunishment || (forcePunishment && slaveryEvent.getIsPunishment())) // Making sure to comply with punishment forcing.
+					{
+						int currentEventWeight = slaveryEvent.getWeight(false);
+						if(currentCandidateWeight < currentEventWeight)
+						{
+							currentCandidateWeight = currentEventWeight;
+							currentCandidate = slaveryEvent;
+						}
+					}
+				}
+			}
+
+			int randomValue = (int )(Math.random() * (GameCharacter.MAX_MINUTES_TO_NEXT_SLAVERY_EVENT - GameCharacter.MIN_MINUTES_TO_NEXT_SLAVERY_EVENT) + 1) + GameCharacter.MIN_MINUTES_TO_NEXT_SLAVERY_EVENT;
+			this.timeOfNextEventCheck = minutesPassed + randomValue;
+			
+			if(currentCandidate != null)
+			{
+				// Warning: We are doing dangerous shit there. Since slavery system is supposed to be an overbearing one, it's safe to do it here, but to other contributors, here's a note - don't do it without a good reason!
+				// To explain - this function below sets the dialogue node.
+				// It can force it in the middle of anything, potentially sequence breaking a quest dialogue.
+				// This has double safeguards in terms of doing it only on minute-changing turn skips *and* only when the player is in a controlled enviroment.
+				// Using it outside of these tight constraints should be avoided.
+				// (Note for Inno: I'm just leaving this as a warning for future contributors :3)
+				Main.game.setContent(new Response("", "", currentCandidate.getTriggerDialogue()));
+			}
+			else
+			{
+				Main.game.setActiveNPC(null);
+			}
+		}
+	}
+	
+	/**
+	 * Returns punishment dialogue node. Doesn't automatically set it like normal; use it for events where player messes up to give them an immediate punishment.
+	 * @return
+	 */
+	public DialogueNodeOld getPunishmentDialogue()
+	{
+		// At 0 weight, an event won't be generated. If all events return 0 weight, then no event happens.
+		Main.game.setActiveNPC((NPC)this);
+		int currentCandidateWeight = 0;
+		PlayerSlaveryEvent currentCandidate = null;
+		
+		if(this.getSlaveryEvents() != null)
```

Comment:
if getSlaveryEvents returns non-null sets, (see below), then you don't need this check.

Reply:
Makes sense